src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/DeliverySessionService.kt

```diff
@@ -170,29 +170,27 @@ class DeliverySessionService(
       attended,
       notifyProbationPractitioner,
     )
-    if (existingAppointment == null) {
-      val appointment = Appointment(
-        id = UUID.randomUUID(),
-        createdBy = authUserRepository.save(updatedBy),
-        createdAt = OffsetDateTime.now(),
-        appointmentTime = appointmentTime,
-        durationInMinutes = durationInMinutes,
-        deliusAppointmentId = deliusAppointmentId,
-        referral = session.referral,
-      )
+
+    val appointment = existingAppointment?.apply {
+      this.appointmentTime = appointmentTime
+      this.durationInMinutes = durationInMinutes
+      this.deliusAppointmentId = deliusAppointmentId
+    } ?: Appointment(
+      id = UUID.randomUUID(),
+      createdBy = authUserRepository.save(updatedBy),
+      createdAt = OffsetDateTime.now(),
+      appointmentTime = appointmentTime,
+      durationInMinutes = durationInMinutes,
+      deliusAppointmentId = deliusAppointmentId,
+      referral = session.referral,
+    )
+    appointmentService.createOrUpdateAppointmentDeliveryDetails(appointment, appointmentDeliveryType, appointmentSessionType, appointmentDeliveryAddress, npsOfficeCode)
+    appointmentRepository.saveAndFlush(appointment)
+    session.appointments.add(appointment)
+    deliverySessionRepository.save(session)
+    return deliverySessionRepository.save(session).also {
       setAttendanceAndBehaviourIfHistoricAppointment(session, appointment, appointmentTime, attended, additionalAttendanceInformation, behaviourDescription, notifyProbationPractitioner, updatedBy)
```

Comment:
i think this would benefit from a comment which explains why it's laid out like this